### PR TITLE
Add FreeBSD support

### PR DESCRIPTION
**Pull request by christopherkobayashi**
With these changes, maestral appears to work on FreeBSD.

FreeBSD and macOS use the same file locking semantics (which makes sense, as macOS uses a modified FreeBSD kernel).

FreeBSD uses the same desktop environments as Linux, so adding FreeBSD to the relevant operating system checks works fine.

FreeBSD has a working kqueue() implementation, so it should be used for the Observer bits.

### CODE REVIEW

**Review by samschott — CHANGES_REQUESTED** [review]
Thank you for the PR, there a appear to be surprisingly few changes necessary :)

My only concern is that using kqueues as a backend is not really tested so far and might lead to unexpected results. I'm not particular familiar with its API and limitations but there have been sufficient differences between FSEvents and inotify for me to be wary of introducing new backends. In addition, the watchdog library has not always been consistent on how it propagates events from different backends to provide uniform behaviour regardless of the platform. Some open questions which I have: 

1. Is there a limit on how many files can be watched with kqueue? Or any performance / memory impact for large folders? Inotify has limits which can be adjusted by the user and we raise appropriate error messages when those are reached. See: https://github.com/samschott/maestral/blob/master/src/maestral/manager.py#L308-L341
2. Are there any other exceptions when starting a kqueue "listener" which we need to handle?
3. What is the behaviour when the watched (Dropbox) folder is deleted or moved?
4. Does watchdog pass along kqueue events consistently with other backends? It took me a couples of days of work with the watchdog folks to get their FSEvents backend working to a degree where it was as reliable as inotify.

Points 3. and 4. are well covered by tests so my preferred approach would be to run the existing test suite with a kqueue backend and see what breaks. GitHub actions do not support FreeBSD but kqueus are available on macOS as well.

File: src/maestral/daemon.py [hunk 1]
```diff
@@ -258,7 +258,11 @@ def locking_pid(self) -> int | None:
                 pid_index = 4
                 flock = struct.pack(fmt, fcntl.F_WRLCK, 0, 0, 0, 0, 0)
 
-            lockdata = fcntl.fcntl(fh.fileno(), fcntl.F_GETLK, flock)
+            try:
+                lockdata = fcntl.fcntl(fh.fileno(), fcntl.F_GETLK, flock)
+            except:
+                return None
```
[the remark this answers is not in the document]
Comment (christopherkobayashi) [reply src/maestral/daemon.py:264]: locking_pid() defaults to Linux-ish semantics; if that's wrong (as it was with FreeBSD), "maestral stop" will exit with an unhandled exception.

Adding a handler here causes "maestral stop" to exit with FAILED, which seems to me to be cleaner and matches the (open to interpretation) documented return values for this function (return PID or None).

We do want to handle only OSError, though -- very valid point.  Would using "except OSError:" be okay?

**Comment by christopherkobayashi** [comment]
I haven't qualitatively tested the kqueue() implementation beyond a simple run test.

maestral runs fine without this backend, and given your past experience with the watchdog library, I think the right thing to do is comment the kqueue() bits out pending me performing an exhaustive test.

I'll amend the pull request to reflect this.

Cheers :)

**Comment by samschott** [comment]
After a bit of poking around, it looks like https://cirrus-ci.org allows running tests on FreeBSD and is free for open source projects. I'll see if can set something up there.

**Pull request closed without merging**